### DISCUSSION TITLE
Build is getting failed after upgrading webpack to v5.

### DISCUSSION THREAD

**Discussion by Arjun0721** [webpack/webpack#16966] [Q&A]
After following the steps to upgrade from webpack v4 to v5, I am ending with this error.

Error: Error on worker #4: Error: Failed to find exported name of node (NgDropdownPanelComponent = /** @class */ (function () {
        function NgDropdownPanelComponent(_renderer, _zone, _panelService, _elementRef, _document) {        
            this._renderer = _renderer;
            this._zone = _zone;
            this._panelService = _panelService;
            this._document = _document;
            this.items = [];
            this.position = 'auto';
            this.virtualScroll = false;
            this.filterValue = null;
            this.update = new i0.EventEmitter();
            this.scroll = new i0.EventEmitter();
            this.scrollToEnd = new i0.EventEmitter();
            this.outsideClick = new i0.EventEmitter();
            this._destroy$ = new rxjs.Subject();
            this._scrollToEndFired = false;
            this._updateScrollHeight = false;
            this._lastScrollPosition = 0;
            this._dropdown = _elementRef.nativeElement;
        }
        Object.defineProperty(NgDropdownPanelComponent.prototype, "currentPosition", {
            get: function () {
                return this._currentPosition;
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(NgDropdownPanelComponent.prototype, "itemsLength", {
            get: function () {
                return this._itemsLength;
            },
            set: function (value) {
                if (value !== this._itemsLength) {
                    this._itemsLength = value;
                    this._onItemsLengthChanged();
                }
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(NgDropdownPanelComponent.prototype, "_startOffset", {
            get: function () {
                if (this.markedItem) {
                    var _a = this._panelService.dimensions, itemHeight = _a.itemHeight, panelHeight = _a.panelHeight;
                    var offset = this.markedItem.index * itemHeight;
                    return panelHeight > offset ? 0 : offset;
                }
                return 0;
            },
            enumerable: false,
            configurable: true
        });
        NgDropdownPanelComponent.prototype.ngOnInit = function () {
            this._select = this._dropdown.parentElement;
            this._virtualPadding = this.paddingElementRef.nativeElement;
            this._scrollablePanel = this.scrollElementRef.nativeElement;
            this._contentPanel = this.contentElementRef.nativeElement;
            this._handleScroll();
            this._handleOutsideClick();
            this._appendDropdown();
            this._setupMousedownListener();
        };
        NgDropdownPanelComponent.prototype.ngOnChanges = function (changes) {
            if (changes.items) {
                var change = changes.items;
                this._onItemsChange(change.currentValue, change.firstChange);
            }
        };
        NgDropdownPanelComponent.prototype.ngOnDestroy = function () {
            this._destroy$.next();
            this._destroy$.complete();
            this._destroy$.unsubscribe();
            if (this.appendTo) {
                this._renderer.removeChild(this._dropdown.parentNode, this._dropdown);
            }
        };
        NgDropdownPanelComponent.prototype.scrollTo = function (option, startFromOption) {
            if (startFromOption === void 0) { startFromOption = false; }
            if (!option) {
                return;
            }
            var index = this.items.indexOf(option);
            if (index < 0 || index >= this.itemsLength) {
                return;
            }
            var scrollTo;
            if (this.virtualScroll) {
                var itemHeight = this._panelService.dimensions.itemHeight;
                scrollTo = this._panelService.getScrollTo(index * itemHeight, itemHeight, this._lastScrollPosition);
            }
            else {
                var item = this._dropdown.querySelector("#" + option.htmlId);
                var lastScroll = startFromOption ? item.offsetTop : this._lastScrollPosition;
                scrollTo = this._panelService.getScrollTo(item.offsetTop, item.clientHeight, lastScroll);   
            }
            if (isDefined(scrollTo)) {
                this._scrollablePanel.scrollTop = scrollTo;
            }
        };
        NgDropdownPanelComponent.prototype.scrollToTag = function () {
            var panel = this._scrollablePanel;
            panel.scrollTop = panel.scrollHeight - panel.clientHeight;
        };
        NgDropdownPanelComponent.prototype.adjustPosition = function () {
            this._updateYPosition();
        };
        NgDropdownPanelComponent.prototype._handleDropdownPosition = function () {
            this._currentPosition = this._calculateCurrentPosition(this._dropdown);
            if (CSS_POSITIONS.includes(this._currentPosition)) {
                this._updateDropdownClass(this._currentPosition);
            }
            else {
                this._updateDropdownClass('bottom');
            }
            if (this.appendTo) {
                this._updateYPosition();
            }
            this._dropdown.style.opacity = '1';
        };
        NgDropdownPanelComponent.prototype._updateDropdownClass = function (currentPosition) {
            var _this = this;
            CSS_POSITIONS.forEach(function (position) {
                var REMOVE_CSS_CLASS = "ng-select-" + position;
                _this._renderer.removeClass(_this._dropdown, REMOVE_CSS_CLASS);
                _this._renderer.removeClass(_this._select, REMOVE_CSS_CLASS);
            });
            var ADD_CSS_CLASS = "ng-select-" + currentPosition;
            this._renderer.addClass(this._dropdown, ADD_CSS_CLASS);
            this._renderer.addClass(this._select, ADD_CSS_CLASS);
        };
        NgDropdownPanelComponent.prototype._handleScroll = function () {
            var _this = this;
            this._zone.runOutsideAngular(function () {
                rxjs.fromEvent(_this.scrollElementRef.nativeElement, 'scroll')
                    .pipe(operators.takeUntil(_this._destroy$), operators.auditTime(0, SCROLL_SCHEDULER))   
                    .subscribe(function (e) {
                    var path = e.path || (e.composedPath && e.composedPath());
                    var scrollTop = !path || path.length === 0 ? e.target.scrollTop : path[0].scrollTop;    
                    _this._onContentScrolled(scrollTop);
                });
            });
        };
        NgDropdownPanelComponent.prototype._handleOutsideClick = function () {
            var _this = this;
            if (!this._document) {
                return;
            }
            this._zone.runOutsideAngular(function () {
                rxjs.merge(rxjs.fromEvent(_this._document, 'touchstart', { capture: true }), rxjs.fromEvent(_this._document, 'mousedown', { capture: true })).pipe(operators.takeUntil(_this._destroy$))
                    .subscribe(function ($event) { return _this._checkToClose($event); });
            });
        };
        NgDropdownPanelComponent.prototype._checkToClose = function ($event) {
            var _this = this;
            if (this._select.contains($event.target) || this._dropdown.contains($event.target)) {
                return;
            }
            var path = $event.path || ($event.composedPath && $event.composedPath());
            if ($event.target && $event.target.shadowRoot && path && path[0] && this._select.contains(path[0])) {
                return;
            }
            this._zone.run(function () { return _this.outsideClick.emit(); });
        };
        NgDropdownPanelComponent.prototype._onItemsChange = function (items, firstChange) {
            this.items = items || [];
            this._scrollToEndFired = false;
            this.itemsLength = items.length;
            if (this.virtualScroll) {
                this._updateItemsRange(firstChange);
            }
            else {
                this._setVirtualHeight();
                this._updateItems(firstChange);
            }
        };
        NgDropdownPanelComponent.prototype._updateItems = function (firstChange) {
            var _this = this;
            this.update.emit(this.items);
            if (firstChange === false) {
                return;
            }
            this._zone.runOutsideAngular(function () {
                Promise.resolve().then(function () {
                    var panelHeight = _this._scrollablePanel.clientHeight;
                    _this._panelService.setDimensions(0, panelHeight);
                    _this._handleDropdownPosition();
                    _this.scrollTo(_this.markedItem, firstChange);
                });
            });
        };
        NgDropdownPanelComponent.prototype._updateItemsRange = function (firstChange) {
            var _this = this;
            this._zone.runOutsideAngular(function () {
                _this._measureDimensions().then(function () {
                    if (firstChange) {
                        _this._renderItemsRange(_this._startOffset);
                        _this._handleDropdownPosition();
                    }
                    else {
                        _this._renderItemsRange();
                    }
                });
            });
        };
        NgDropdownPanelComponent.prototype._onContentScrolled = function (scrollTop) {
            if (this.virtualScroll) {
                this._renderItemsRange(scrollTop);
            }
            this._lastScrollPosition = scrollTop;
            this._fireScrollToEnd(scrollTop);
        };
        NgDropdownPanelComponent.prototype._updateVirtualHeight = function (height) {
            if (this._updateScrollHeight) {
                this._virtualPadding.style.height = height + "px";
                this._updateScrollHeight = false;
            }
        };
        NgDropdownPanelComponent.prototype._setVirtualHeight = function () {
            if (!this._virtualPadding) {
                return;
            }
            this._virtualPadding.style.height = "0px";
        };
        NgDropdownPanelComponent.prototype._onItemsLengthChanged = function () {
            this._updateScrollHeight = true;
        };
        NgDropdownPanelComponent.prototype._renderItemsRange = function (scrollTop) {
            var _this = this;
            if (scrollTop === void 0) { scrollTop = null; }
            if (scrollTop && this._lastScrollPosition === scrollTop) {
                return;
            }
            scrollTop = scrollTop || this._scrollablePanel.scrollTop;
            var range = this._panelService.calculateItems(scrollTop, this.itemsLength, this.bufferAmount);  
            this._updateVirtualHeight(range.scrollHeight);
            this._contentPanel.style.transform = "translateY(" + range.topPadding + "px)";
            this._zone.run(function () {
                _this.update.emit(_this.items.slice(range.start, range.end));
                _this.scroll.emit({ start: range.start, end: range.end });
            });
            if (isDefined(scrollTop) && this._lastScrollPosition === 0) {
                this._scrollablePanel.scrollTop = scrollTop;
                this._lastScrollPosition = scrollTop;
            }
        };
        NgDropdownPanelComponent.prototype._measureDimensions = function () {
            var _this = this;
            if (this._panelService.dimensions.itemHeight > 0 || this.itemsLength === 0) {
                return Promise.resolve(this._panelService.dimensions);
            }
            var _a = __read(this.items, 1), first = _a[0];
            this.update.emit([first]);
            return Promise.resolve().then(function () {
                var option = _this._dropdown.querySelector("#" + first.htmlId);
                var optionHeight = option.clientHeight;
                _this._virtualPadding.style.height = optionHeight * _this.itemsLength + "px";
                var panelHeight = _this._scrollablePanel.clientHeight;
                _this._panelService.setDimensions(optionHeight, panelHeight);
                return _this._panelService.dimensions;
            });
        };
        NgDropdownPanelComponent.prototype._fireScrollToEnd = function (scrollTop) {
            var _this = this;
            if (this._scrollToEndFired || scrollTop === 0) {
                return;
            }
            var padding = this.virtualScroll ?
                this._virtualPadding :
                this._contentPanel;
            if (scrollTop + this._dropdown.clientHeight >= padding.clientHeight - 1) {
                this._zone.run(function () { return _this.scrollToEnd.emit(); });
                this._scrollToEndFired = true;
            }
        };
        NgDropdownPanelComponent.prototype._calculateCurrentPosition = function (dropdownEl) {
            if (this.position !== 'auto') {
                return this.position;
            }
            var selectRect = this._select.getBoundingClientRect();
            var scrollTop = document.documentElement.scrollTop || document.body.scrollTop;
            var offsetTop = selectRect.top + window.pageYOffset;
            var height = selectRect.height;
            var dropdownHeight = dropdownEl.getBoundingClientRect().height;
            if (offsetTop + height + dropdownHeight > scrollTop + document.documentElement.clientHeight) {  
                return 'top';
            }
            else {
                return 'bottom';
            }
        };
        NgDropdownPanelComponent.prototype._appendDropdown = function () {
            if (!this.appendTo) {
                return;
            }
            this._parent = document.querySelector(this.appendTo);
            if (!this._parent) {
                throw new Error("appendTo selector " + this.appendTo + " did not found any parent element");            }
            this._updateXPosition();
            this._parent.appendChild(this._dropdown);
        };
        NgDropdownPanelComponent.prototype._updateXPosition = function () {
            var select = this._select.getBoundingClientRect();
            var parent = this._parent.getBoundingClientRect();
            var offsetLeft = select.left - parent.left;
            this._dropdown.style.left = offsetLeft + 'px';
            this._dropdown.style.width = select.width + 'px';
            this._dropdown.style.minWidth = select.width + 'px';
        };
        NgDropdownPanelComponent.prototype._updateYPosition = function () {
            var select = this._select.getBoundingClientRect();
            var parent = this._parent.getBoundingClientRect();
            var delta = select.height;
            if (this._currentPosition === 'top') {
                var offsetBottom = parent.bottom - select.bottom;
                this._dropdown.style.bottom = offsetBottom + delta + 'px';
                this._dropdown.style.top = 'auto';
            }
            else if (this._currentPosition === 'bottom') {
                var offsetTop = select.top - parent.top;
                this._dropdown.style.top = offsetTop + delta + 'px';
                this._dropdown.style.bottom = 'auto';
            }
        };
        NgDropdownPanelComponent.prototype._setupMousedownListener = function () {
            var _this = this;
            this._zone.runOutsideAngular(function () {
                rxjs.fromEvent(_this._dropdown, 'mousedown')
                    .pipe(operators.takeUntil(_this._destroy$))
                    .subscribe(function (event) {
                    var target = event.target;
                    if (target.tagName === 'INPUT') {
                        return;
                    }
                    event.preventDefault();
                });
            });
        };
        return NgDropdownPanelComponent;
    }())) in 'C:/Users/U6070673/IdeaProjects/incites-frontend/node_modules/@ng-select/ng-select/bundles/ng-select-ng-select.umd.js'.
    at Object.findExportedNameOfNode (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\src\ngtsc\imports\src\find_export.js:56:19)
    at LogicalProjectStrategy.emit (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\src\ngtsc\imports\src\emitter.js:256:38)
    at ReferenceEmitter.emit (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\src\ngtsc\imports\src\emitter.js:72:44)
    at Object.toR3Reference (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\src\ngtsc\annotations\src\util.js:247:18)
    at NgModuleDecoratorHandler._toR3Reference (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\src\ngtsc\annotations\src\ng_module.js:535:31)
    at C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\src\ngtsc\annotations\src\ng_module.js:265:83
    at Array.map (<anonymous>)
    at NgModuleDecoratorHandler.analyze (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\src\ngtsc\annotations\src\ng_module.js:265:48)
    at NgccTraitCompiler.TraitCompiler.analyzeTrait (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\src\ngtsc\transform\src\compilation.js:421:40)
    at analyze (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\src\ngtsc\transform\src\compilation.js:371:58)
    at ClusterMaster.onWorkerMessage (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:195:27)
    at C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:55:95
    at ClusterMaster.<anonymous> (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:293:57)
    at step (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\node_modules\tslib\tslib.js:144:27)
    at Object.next (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\node_modules\tslib\tslib.js:125:57)
    at C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\node_modules\tslib\tslib.js:118:75
    at new Promise (<anonymous>)
    at Object.__awaiter (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\node_modules\tslib\tslib.js:114:16)
    at EventEmitter.<anonymous> (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@angular\compiler-cli\ngcc\src\execution\cluster\master.js:287:32)
    at EventEmitter.emit (node:events:527:28)
[webpack-cli] Error: NGCC failed.
    at NgccProcessor.process (C:\Users\U6070673\IdeaProjects\incites-frontend\node_modules\@ngtools\webpack\src\ngcc_processor.js:129:19)

**Comment by snitin315** [comment]
Thank you for creating this issue. However, issues need to follow one of our templates so that we can clearly understand your particular circumstances.

Please help us help you by recreating the issue using one of our templates.
